File: dotnetConversion/src/Aion.LoginServer/Model/GameServerInfo.cs
using System.Collections.Concurrent;

namespace Aion.LoginServer.Model;

public sealed class GameServerInfo
{
	private readonly ConcurrentDictionary<int, Account> _accountsOnGameServer = new();

	public GameServerInfo(byte id, string ipMask, string password)
	{
		Id = id;
		IpMask = ipMask;
		Password = password;
	}

	public byte Id { get; }

	public string IpMask { get; }

	public string Password { get; }

	public byte[] Ip { get; private set; } = { 0, 0, 0, 0 };

	public ushort Port { get; private set; }

	public byte MinAccessLevel { get; private set; }

	public int MaxPlayers { get; private set; }

	public bool IsOnline { get; private set; }

	public int CurrentPlayers => _accountsOnGameServer.Count;

	public bool IsFull => CurrentPlayers >= MaxPlayers;

	public void MarkOnline(byte[] ip, ushort port, byte minAccessLevel, int maxPlayers)
	{
		Ip = ip;
		Port = port;
		MinAccessLevel = minAccessLevel;
		MaxPlayers = maxPlayers;
		IsOnline = true;
	}

	public void MarkOffline()
	{
		IsOnline = false;
		_accountsOnGameServer.Clear();
	}

	public bool IsAccountOnGameServer(int accountId) => _accountsOnGameServer.ContainsKey(accountId);

	public void AddAccount(Account account) => _accountsOnGameServer[account.Id] = account;

	public Account? RemoveAccount(int accountId)
	{
		_accountsOnGameServer.TryRemove(accountId, out var account);
		return account;
	}

	public Account? GetAccount(int accountId)
	{
		_accountsOnGameServer.TryGetValue(accountId, out var account);
		return account;
	}
}

File: dotnetConversion/tools/java-login-crypto-vectors/com/aionemu/loginserver/GameServerInfo.java
package com.aionemu.loginserver;

public final class GameServerInfo {
	private final byte id;
	private final byte[] ip;
	private final int port;
	private final int currentPlayers;
	private final int maxPlayers;
	private final boolean online;

	public GameServerInfo(byte id, byte[] ip, int port, int currentPlayers, int maxPlayers, boolean online) {
		this.id = id;
		this.ip = ip;
		this.port = port;
		this.currentPlayers = currentPlayers;
		this.maxPlayers = maxPlayers;
		this.online = online;
	}

	public byte getId() {
		return id;
	}

	public byte[] getIp() {
		return ip;
	}

	public int getPort() {
		return port;
	}

	public int getCurrentPlayers() {
		return currentPlayers;
	}

	public int getMaxPlayers() {
		return maxPlayers;
	}

	public boolean isOnline() {
		return online;
	}
}
